x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/core/tree/Node.java

```diff
@@ -367,31 +368,36 @@ public List<Object> nodeProperties() {
     }
 
     public String nodeString() {
+        return nodeString(propertiesToString(true /* skipIfChild */));
+    }
+
+    /**
+     * Like the above, but ensures consistent tree rendering for golden tests. Overrides of this method should avoid adding information
+     * that can change between repeated test runs, such as object identities or pseudo-random generated names.
+     */
+    public String goldenTestNodeString() {
```

Comment:
Why are you modifying all these core classes of ESQL for something that is inherently testing related?
This is production code that should be transparent to whatever testing infrastructure needs.